src/Microsoft.Azure.WebJobs.Host/Bindings/ExecutionContext/ExecutionContext.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Provides context information for job function invocations.
+    /// </summary>
+    public class ExecutionContext
+    {
+        /// <summary>
+        /// Gets or sets the job function invocation ID
+        /// </summary>
+        public Guid InvocationId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the function being invoked
+        /// </summary>
+        public string FunctionName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the function directory. This is <see cref="FunctionAppDirectory"/> concatenated with <see cref="FunctionName"/>
+        /// </summary>
+        public string FunctionDirectory { get; set; }
```

Comment:
Why do we have this at all, if it is just a concatenation of two other properties on this type?

Reply:
This is not clear to customers using functions and a more convenient way to get the current directory. This is existing functionality that is being ported over.